### DISCUSSION TITLE
Custom Scheme IResourceHandler IRequest accept ajax post request, IRequest is null,ajax postdata have data

### DISCUSSION THREAD

**Discussion by PingXD** [cefsharp/CefSharp#3860] [General]
request:
![image](https://user-images.githubusercontent.com/48038690/138537707-ca28dd35-188f-41f0-9c14-0709c069389c.png)
"xbro"  is my Custom Scheme,

in IResourceHandler.open  IRequest parameter,this parameter  POSTData  attribute is null
![image](https://user-images.githubusercontent.com/48038690/138537829-6837cf94-54a1-4c87-a838-50e16e650fcd.png)

**Comment by PingXD** [comment]
if i use Html5  native  submit form data this attribute can accept formdata, if  use ajax it is faild

**Comment by amaitland** [comment]
No images of code.

Start by answering some basic questions see https://github.com/cefsharp/CefSharp/wiki/Questions-&-Support#when-asking-a-question-please-include-the-following

> "xbro" is my Custom Scheme,

How did you register your scheme exactly?

**Reply by PingXD** [reply]
in Cef.Initialize  us 
```
  cefSettings.RegisterScheme(new CefCustomScheme()
            {
                SchemeName = ResourceSchemeHandlerFactory.SchemeName,
                SchemeHandlerFactory = new ResourceSchemeHandlerFactory(), 
            });

```

```
 public class ResourceSchemeHandlerFactory : ISchemeHandlerFactory
    {
        public static string SchemeName { get { return "xbro"; } }


        public IResourceHandler Create(IBrowser browser, IFrame frame, string schemeName, IRequest request)
        {
            return new ResourceSchemeHandler();
        }
    }

```



ResourceSchemeHandler   is  IResourceHandler
override  ProcessRequestAsync

**Reply by PingXD** [reply]
about image of code i am sorry

**Reply by amaitland** [reply]
> about image of code i am sorry

Please provide your actual code, no images.

You can try enabling fetch for your custom scheme.

https://cefsharp.github.io/api/94.4.x/html/P_CefSharp_CefCustomScheme_IsFetchEnabled.htm

**Comment by jankurianski** [comment]
https://bitbucket.org/chromiumembedded/cef/wiki/GeneralUsage.md#markdown-header-request-handling

> Use the HTTP scheme instead of a custom scheme to avoid a range of potential issues.

I initially tried using a custom scheme in my application too but found lots of issues.  It is much easier for you if you use the http scheme and register your own domain.  First you could try this with your custom scheme to see if it makes it behave more like http:

```
cefSettings.RegisterScheme(new CefCustomScheme {
    SchemeName = ResourceSchemeHandlerFactory.SchemeName,
    SchemeHandlerFactory = new ResourceSchemeHandlerFactory(), 
    IsStandard = true,
    IsCorsEnabled = true
});
```

But if this doesn't work, I would recommend you use `SchemeName` of `http` and set the `DomainName` property to `xbro`, so you would instead use URLs like `http://xbro/`.
